### DISCUSSION TITLE
Next.Js 15 SWC Support

### DISCUSSION THREAD

**Discussion by shawnmclean** [founded-labs/react-native-reusables#165] [Q&A]
Has anyone gotten this code, or even up to NativeWind to work with nextjs 14 or higher?

Mind sharing a working repo?

Does it work with HMR if in a shared lib package?

**Comment by megacherry** [comment]
yes, but some styles are not applied. in summary i think at this point it's not really useable with nextjs.

**Reply by mrzachnugent** [reply]
@megacherry Which styles are not applied? I've only tried it react-native-web, I've never tried it with nextjs

**Reply by shawnmclean** [reply]
HoverCard doesn't work well for me:

Demo page, hover one of the list of books: https://www.sovoli.com/shawn/shelves/wardrobe-bottom-left-shelf

Code: https://github.com/shawnmclean/sovoli/blob/main/packages/ui/src/components/BookHoverCard/BookHoverCard.tsx

Note: repo tries to follow t3 turbo template. So the same issue may happen there.

Maybe a good idea would be to implement this universal in t3 since they already have next + expo running in a monorepo. People have been asking in the issues to share components between the 2.

**Reply by megacherry** [reply]
i commented here: https://github.com/roninoss/rn-primitives/issues/17

the same issue exists with avatar (classNames not applied). i looks like a general problem.

**Comment by atanaskanchev** [comment]
I've tried to implement the components in a turbo monorepo with expo and next apps, but some styles are not being applied even in client components. Shall we convert this discussion to an issue @mrzachnugent ?

**Reply by mrzachnugent** [reply]
This is related to `@rn-primitives`. There is a existing issue https://github.com/roninoss/rn-primitives/issues/17#issuecomment-2242821969 , it is still under investivation.

**Comment by mrzachnugent** [comment]
Primitives now work with classNames in nextjs 🎉

Update your primitives to their latest version and try again.
